crates/cxx-qt-gen/src/parser/mod.rs

```diff
@@ -21,15 +21,71 @@ use crate::{
     naming::TypeNames,
     syntax::{expr::expr_to_string, path::path_compare_str, safety::Safety},
 };
+use convert_case::Case;
 use cxxqtdata::ParsedCxxQtData;
 use std::collections::BTreeMap;
 use syn::{
     punctuated::Punctuated,
     spanned::Spanned,
     token::{Brace, Semi},
-    Attribute, Error, ForeignItemFn, Ident, Item, ItemMod, Meta, Result, Token, Visibility,
+    Attribute, Error, Expr, ForeignItemFn, Ident, Item, ItemMod, Meta, Result, Token, Visibility,
 };
 
+#[derive(Copy, Clone)]
+pub struct CaseConversion {
+    pub cxx: Option<Case>,
+    pub rust: Option<Case>,
+}
+
+/// Used to match the auto_case attributes and turn it into a Case to convert to
+fn meta_to_case(attr: &Attribute, default: Case) -> Result<Case> {
+    match &attr.meta {
+        Meta::Path(_) => Ok(default),
+        Meta::NameValue(case) => match &case.value {
+            Expr::Path(expr_path) => match expr_path.path.require_ident()?.to_string().as_str() {
```

Comment:
You might be able to remove the `.to_string().as_str()`, as `Ident` can be compared with a str in certain places? Unsure if that works here?

```
impl<T> PartialEq<T> for Ident
where
    T: AsRef<str> + ?Sized,
```